### PR TITLE
chore: add types to saveAndReturn

### PR DESCRIPTION
**Pull request by gunar**

### CODE REVIEW

**Comment by gunar** [comment]
Works on Pizza

```
root@1201:~/planx-new# docker-compose logs seed-database
Attaching to planx-new_seed-database_1
seed-database_1  | {"level":"info","msg":"Help us improve Hasura! The cli collects anonymized usage stats which\nallow us to keep improving Hasura at warp speed. To opt-out or read more,\nvisit https://hasura.io/docs/latest/graphql/core/guides/telemetry.html\n","time":"2022-10-12T14:27:59Z"}
seed-database_1  | {"level":"info","msg":"Applying seeds...","time":"2022-10-12T14:27:59Z"}
seed-database_1  | {"level":"info","msg":"Seeds planted","time":"2022-10-12T14:27:59Z"}
seed-database_1  |
seed-database_1  | > seed-database@ upsert-flows /scripts
seed-database_1  | > node --experimental-specifier-resolution=node ./upsert-production-flows
seed-database_1  |
seed-database_1  | Fetching teams and flows...
seed-database_1  | Inserting flows and teams...
seed-database_1  | Fetching published flows...
seed-database_1  | Production flows and teams inserted successfully.
root@1201:~/planx-new#
```

File: api.planx.uk/types.ts [hunk 1]
```diff
@@ -12,3 +12,104 @@ export interface Flow {
...
+  updated_at: string;
+  has_user_saved: boolean;
+  flow: {
+    slug: string;
+  };
+}
+
+export interface NotifyConfig {
+  personalisation: {
+    has_user_saved?: boolean | undefined;
```
Comment (DafyddLlyr) [inline api.planx.uk/types.ts:60]: I think this is a mistake - `has_user_saved` is a column on the lowcal_sessions table whilst `NotifyConfig.personalisation` represents the variables that can be passed into an email sent via Notify.

File: api.planx.uk/types.ts [hunk 2]
```diff
@@ -12,3 +12,104 @@ export interface Flow {
...
+    emailReplyToId: string;
+  };
+}
+
+export interface Passport {
+  data?: Record<string, any>;
+}
+
+export interface LowCalSession {
+  govUkPayment: any;
```
Comment (DafyddLlyr) [inline api.planx.uk/types.ts:42]: This looks like a duplicate - I think only the `govUkPayment` nested within `data` should be correct.

**Review by DafyddLlyr — CHANGES_REQUESTED** [review]
Thanks so much for picking this up - great to see! A few very minor comments - let me know if you have any issues and I'll be happy to take a look.

File: api.planx.uk/types.ts [hunk 3]
```diff
@@ -12,3 +12,104 @@ export interface Flow {
...
+    helpOpeningHours: string;
+    emailReplyToId: string;
+  };
+}
+
+export interface Passport {
+  data?: Record<string, any>;
+}
+
+export interface LowCalSession {
```
Comment (jessicamcinchak) [inline api.planx.uk/types.ts:41]: Can you also please add `email: string;` & `flow_id: string;` to this type as they're required fields?

File: api.planx.uk/types.ts [hunk 4]
```diff
@@ -12,3 +12,104 @@ export interface Flow {
...
+    serviceLink?: string;
+    serviceName?: string;
+    teamName: string;
+  };
+  reference?: any;
+  emailReplyToId?: any;
+}
+
+// https://docs.payments.service.gov.uk/making_payments/#receiving-the-api-response
+export interface GovUKPayment {
```
Comment (jessicamcinchak) [inline api.planx.uk/types.ts:79]: :+1:

one question: in the proxy response of our `/pay/:localAuthority/:paymentId` API endpoint, we'll return a filtered response that is only payment_id, amount, and state - should we mark other fields as optional here to account for that?

**Review by jessicamcinchak — COMMENTED** [review]
thanks for starting this! added a couple comments & suggetions for tweaks to type definitions, but overall looking good :+1:

**Comment by gunar** [comment]
Thanks for the review! I think I've now addressed all the feedback :+1:

**Review by DafyddLlyr — APPROVED** [review]
Perfect 👌

**Pull request merged**